services/server/src/server/cli.ts

```diff
@@ -0,0 +1,241 @@
+import path from "path";
+// First env vars need to be loaded before config
+import dotenv from "dotenv";
+dotenv.config({ path: path.resolve(__dirname, "..", "..", ".env") });
+// Make sure config is relative to index.ts and not where the server is run from
+process.env["NODE_CONFIG_DIR"] = path.resolve(__dirname, "..", "config");
+import config from "config";
+import swaggerUi from "swagger-ui-express";
+import yamljs from "yamljs";
+import expressSession from "express-session";
+import createMemoryStore from "memorystore";
+import { Pool } from "pg";
+import genFunc from "connect-pg-simple";
+
+// local imports
+import logger from "../common/logger";
+import { sourcifyChainsMap } from "../sourcify-chains";
+import { Server } from "./server";
+import { ChainRepository } from "../sourcify-chain-repository";
+import { ISolidityCompiler } from "@ethereum-sourcify/lib-sourcify";
+import { SolcLambdaWithLocalFallback } from "./services/compiler/lambda-with-fallback/SolcLambdaWithLocalFallback";
+import { SolcLocal } from "./services/compiler/local/SolcLocal";
+import session from "express-session";
+
+
+// Supported Chains
+
+const chainRepository = new ChainRepository(sourcifyChainsMap);
+
+logger.info("SourcifyChains.Initialized", {
+  supportedChainsCount: chainRepository.supportedChainsArray.length,
+  allChainsCount: chainRepository.sourcifyChainsArray.length,
+  supportedChains: chainRepository.supportedChainsArray.map((c) => c.chainId),
+  allChains: chainRepository.sourcifyChainsArray.map((c) => c.chainId),
+});
+
+// Solidity Compiler
+
+
+const repoPath = (config.get("solcRepo") as string) || path.join("/tmp", "solc-repo");
+const soljsonRepo = (config.get("solJsonRepo") as string) || path.join("/tmp", "soljson-repo");
+
+if (
+  process.env.AWS_REGION === undefined ||
+  process.env.AWS_ACCESS_KEY_ID === undefined ||
+  process.env.AWS_SECRET_ACCESS_KEY === undefined
+) {
+  throw new Error(
+    "AWS credentials not set. Please set them to run the compiler on AWS Lambda.",
+  );
+}
+
+let selectedSolidityCompiler: ISolidityCompiler;
+if (config.get("lambdaCompiler.enabled")) {
+  logger.info("Using lambda solidity compiler with local fallback");
```

Comment:
This throwing when no credentials found should only be if `lambdaCompiler.enabled`, so move it inside if `lambdaCompiler.enabled`.

```suggestion
let selectedSolidityCompiler: ISolidityCompiler;
if (config.get("lambdaCompiler.enabled")) {
  logger.info("Using lambda solidity compiler with local fallback");

  if (
    process.env.AWS_REGION === undefined ||
    process.env.AWS_ACCESS_KEY_ID === undefined ||
    process.env.AWS_SECRET_ACCESS_KEY === undefined
  ) {
    throw new Error(
      "AWS credentials not set. Please set them to run the compiler on AWS Lambda.",
    );
  }
```

Reply:
Fixed in [6447916](https://github.com/ethereum/sourcify/pull/1637/commits/6447916591a4b914ce9ae43588a931bcea8b0d7a)